ckanext/aircan/logic/action.py

```
@@ -0,0 +1,178 @@
+      - message (optional)
+      - dag_run_id (optional)
+      - state (optional)
+      - clear_logs (optional, bool-ish)
+      - key (optional, default: "pipeline")
+      - type (optional, default: "info"; if "error" -> marks task failed)
+      - last_updated (optional)
+    """
+    resource_id = data_dict.get("resource_id")
+
+    if not resource_id:
+        raise p.toolkit.ValidationError({"resource_id": ["Missing resource_id"]})
```

Ruff (0.14.10)

Undefined name `p`

(F821)